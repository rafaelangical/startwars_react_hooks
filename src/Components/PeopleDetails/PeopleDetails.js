/* eslint-disable no-nested-ternary */
/* eslint-disable jsx-a11y/img-redundant-alt */
/* eslint-disable no-unused-vars */
/* eslint-disable react/no-array-index-key */
/* eslint-disable no-unused-expressions */
import React, { useState, useEffect } from "react";
import Loading from "../Loading/Loading";
import api from "../../services/api";

export default function PeopleDetails(props) {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);
  const [data, setData] = useState(null);
  const { people, show } = props;

  useEffect(() => {
    const planets =
      people && people.starships
        ? people.starships.map(item =>
            api
              .get(item)
              .then(resp => resp.data)
              .catch(err => setError(true))
          )
        : null;
    show &&
      Promise.all(planets)
        .then(values => {
          setLoading(false);
          setData(values);
          return values;
        })
        .catch(err => {
          setError(true);
        });
    return setData;
  }, [people, people.starships, show]);
  return (
    people && (
      <div className="Modal-main">
        <div className="modal">
          {error ? (
            <h3>Error planets :(</h3>
          ) : !error && loading ? (
            <Loading />
          ) : (
            <>
              <img src="#" alt="image-person" />
              <div className="divider" />
              <h2>{people.name}</h2>
              <p>
                Altura:
                <span>{people.height}</span>
              </p>
              <p>
                Cor dos olhos:
                <span>{people.eye_color}</span>
              </p>
              <p>
                GÃªnero:
                <span>{people.gender}</span>
              </p>
              <p>
                Peso:
                <span>{people.mass}</span>
              </p>
              <div className="divider" />
              {people.starships.length > 0 && <h4>Starchips</h4>}
              {data &&
                data.map((item, key) => (
                  <div className="planet-info" key={key}>
                    <p>
                      Nome:
                      <span>{item.name}</span>
                    </p>
                    <p>
                      Fabricante:
                      <span>{item.manufacturer}</span>
                    </p>
                    <p>
                      Modelo:
                      <span>{item.model}</span>
                    </p>
                    <p>
                      Tamanho:
                      <span>{item.length}</span>
                    </p>
                    <p>
                      Capacidade:
                      <span>{item.cargo_capacity}</span>
                    </p>
                  </div>
                ))}
            </>
          )}
        </div>
      </div>
    )
  );
}
