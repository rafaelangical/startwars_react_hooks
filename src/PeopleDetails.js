import React from "react";

export default function PeopleDetails(props) {
  const { people } = props;
  return (
    people && (
      <div className="modal">
        <img src="#" />
        <h2>{people.name}</h2>
        <span>Height: {people.height}</span>
        <span>Color Eye: {people.eye_color}</span>
        <span>Gender: {people.gender}</span>
        <span>Mass: {people.mass}</span>
      </div>
    )
  );
}
