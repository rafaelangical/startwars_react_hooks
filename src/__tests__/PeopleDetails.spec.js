/* eslint-disable prefer-destructuring */
import React from "react";
import renderer from "react-test-renderer";
import PeopleDetails from "../Components/PeopleDetails/PeopleDetails";

it("renders correctly but not visual effect", () => {
  const people = {};
  const tree = renderer
    .create(<PeopleDetails show={false} people={people} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with data", () => {
  const people = {
    name: "axzxx",
    height: 20,
    eye_color: "blue",
    starships: ["http://url1", "http://url2"],
  };
  const tree = renderer.create(<PeopleDetails show people={people} />).toJSON();
  expect(tree).toMatchSnapshot();
});
