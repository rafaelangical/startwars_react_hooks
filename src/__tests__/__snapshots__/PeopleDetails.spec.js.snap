// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly but not visual effect 1`] = `
<div
  className="Modal-main"
>
  <div
    className="modal"
  >
    <div
      className="lds-ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`renders correctly with data 1`] = `
<div
  className="Modal-main"
>
  <div
    className="modal"
  >
    <div
      className="lds-ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
